File: backend/ferbook/src/main/java/hr/fer/ppp/ferbook/api/model/Comment.java
package hr.fer.ppp.ferbook.api.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.sql.Timestamp;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Comment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int ID;

    @ManyToOne
    private User user;

    @ManyToOne
    @JsonIgnore
    @JoinColumn
    private Activity activity;

    @NotNull
    private String content;

    @CreationTimestamp
    private Timestamp createdAt;
}

File: frontend/stores/models/comment.js
import { Activity } from "./activity";
import { User } from "./user";

export class Comment {
  constructor(data) {
    this.id = data.id;
    this.content = data.content;
    this.user = data.user && new User(data.user);
    this.activity = data.activity && new Activity(data.activity);
    this.createdAt = data.createdAt;
  }

  serialize() {
    return {
      id: this.id,
      content: this.content,
      activityID: this.activity.id,
    }
  }
}
